### PR TITLE
Remove backward compatibility with XML 15.

### PR DESCRIPTION
**Pull request by wvreeven**

### CODE REVIEW

File: tests/test_siglent_ssa3000x_spectrum_analyzer_data_client.py [hunk 1]
```diff
@@ -85,23 +84,18 @@ def get_data_client(self) -> csc.SiglentSSA3000xSpectrumAnalyzerDataClient:
...
-                        csc.SiglentSSA3000xSpectrumAnalyzerDataClient.start_frequency
-                    )
-                    assert telemetry.stopFrequency == pytest.approx(
-                        csc.SiglentSSA3000xSpectrumAnalyzerDataClient.stop_frequency
-                    )
-                    assert np.amax(telemetry.spectrum) <= 0.0
-                    assert np.amin(telemetry.spectrum) >= -100.0
-                finally:
-                    await data_client.stop()
+            await data_client.start()
```
Comment (r-owen) [inline tests/test_siglent_ssa3000x_spectrum_analyzer_data_client.py:89]: This suggests that the create_data_client method could (and probably should) be an async context manager. That saves you the bother of awaiting start, try/finally/stop.

Or do you need to be able to call it as not-an-async-manager as well?

**Review by r-owen — APPROVED** [review]
Looks very good. A few minor comments.

File: python/lsst/ts/ess/csc/data_client/lightning_data_client.py [hunk 2]
```diff
@@ -229,11 +224,7 @@ async def process_efm100c_telemetry(
         if not topic_kwargs:
             return
 
-        # TODO DM-38363 Remove the line containing "if" as soon as XML 16 has
-        #  been released and leave the line setting "location" in place.
-        if self.has_location:
-            topic_kwargs["location"] = device_configuration.location
-
+        topic_kwargs["location"] = device_configuration.location
```
[the remark this answers is not in the document]
Comment (wvreeven) [reply python/lsst/ts/ess/csc/data_client/lightning_data_client.py:227]: Thanks for the suggestion. I looked into this and this doesn't seem to be practical. The config of this data client has two devices for which the order in the config list depends on the configuration. It is possible to loop over all devices, test for a specific sensor type and, based on that, set the `location` field but I feel that that adds too much clutter. I prefer to leave it as is.

File: python/lsst/ts/ess/csc/data_client/lightning_data_client.py [hunk 3]
```diff
@@ -376,11 +366,8 @@ async def process_ld250_noise_or_status(
             "closeAlarmStatus": close_alarm_status,
             "severeAlarmStatus": severe_alarm_status,
             "heading": heading,
+            "location": device_configuration.location,
```
[the remark this answers is not in the document]
Comment (wvreeven) [reply python/lsst/ts/ess/csc/data_client/lightning_data_client.py:369]: See my previous comment.

File: python/lsst/ts/ess/csc/data_client/siglent_ssa3000x_spectrum_analyzer_data_client.py [hunk 4]
```diff
@@ -227,7 +223,9 @@ async def read_data(self) -> None:
             raise
         raw_data = read_bytes.decode().strip()
         raw_data_items = raw_data.split(",")
-        data = [float(i.strip()) for i in raw_data_items]
+        # The data from the spectrum analyzer end in a "," so the last item
+        # will be an empty string and needs to be dropped.
```
[the remark this answers is not in the document]
Comment (wvreeven) [reply python/lsst/ts/ess/csc/data_client/siglent_ssa3000x_spectrum_analyzer_data_client.py:227]: I adopted the test and removing it conditionally, though I did correct your code example to
```
if raw_data_items[-1] == "":
    del raw_data_items[-1]
```

File: tests/test_ess_csc.py [hunk 5]
```diff
@@ -720,23 +720,20 @@ async def test_spectrum_analyzer_data_client_loses_connection(self) -> None:
...
+                remote=self.remote, state=salobj.State.ENABLED
+            )
+            await self.assert_next_summary_state(salobj.State.ENABLED)
+            assert len(self.csc.data_clients) == 1
+            assert self.csc.data_clients[0].mock_data_server is not None
+            assert self.csc.data_clients[0].mock_data_server.connected
+
+            # The CSC should go to FAULT state because of the long interval
+            # between reading consecutive data.
+            await self.assert_next_summary_state(salobj.State.FAULT)
```
[the remark this answers is not in the document]
Comment (wvreeven) [reply tests/test_ess_csc.py:739]: Thanks for the suggestion. I found that all but one calls to `assert_next_summary_state` didn't have a timeout so I added one to all.

File: tests/test_siglent_ssa3000x_spectrum_analyzer_data_client.py (hunk 1 above)
Comment (wvreeven) [reply tests/test_siglent_ssa3000x_spectrum_analyzer_data_client.py:89]: As discussed on Slack, this will be done in DM-38876.

**Pull request merged**